irohad/main/application.cpp

```diff
@@ -204,6 +212,17 @@ bool Irohad::restoreWsv() {
       });
 }
 
+bool Irohad::resetPeers(
+    shared_model::interface::types::PeerList alternative_peers) {
+  storage->resetPeers();
+  for (const auto &peer : alternative_peers) {
+    if (not storage->insertPeer(*peer)) {
+      return false;
```

Comment:
1. The return value is not utilized anywhere.
2. The error message is not informative and misleading in some cases. For example, some of the peers were inserted successfully, the last one was not. In that case, the error message will be shown and the node can continue its proper initialization (depends on network configuration and block store state).

Thus the logging needs to be reworked.
I suggest logging each successfully inserted peer using info level.
Yes, here info level suits perfectly.

Reply:
For 1 - I think it would be resolved in https://github.com/hyperledger/iroha/pull/41
For 2 my opinion is that in every possible problem on an init phase (even for ones we can recovery from) it's better to just crash. It will so one day - see p. 1.
I'm not sure I understand the rest part well - there are already info log messages for every insertion.